### ISSUE TITLE
`sol!` tracking issue

### ISSUE THREAD

**Issue by DaniPopes** [alloy-rs/core#74]
This is a tracking issue for missing `sol!` proc macro (and corresponding `sol-types`, `syn-solidity`) features.

## Blockers

- [x] #68
- [x] #38
- [x] #55
- [x] #73

## Nice-to-haves

- [x] #36
- [x] #37
- [x] #39
- [x] #82
- [x] #163
- [x] #157, #184
- [x] #254
- [ ] #261

**Comment by mahmudsudo** [comment]
can i take on this ?

**Comment by benluelo** [comment]
Is it planned to support resolving imports from within the sol macro? Currently `sol!` is largely unusable for any non-trivial contracts, since all imports have to be re-specified manually..

**Comment by DaniPopes** [comment]
> Is it planned to support resolving imports from within the sol macro? Currently sol! is largely unusable for any non-trivial contracts, since all imports have to be re-specified manually.

@benluelo

No, that would be basically embedding a compiler frontend into the macro at that point which is just not worth it.

You should use [JSON ABI input](https://docs.rs/alloy-sol-macro/latest/alloy_sol_macro/macro.sol.html#json-abi) to pass in ABI generated from other tools, or write a simple interface inline.
